hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/symmetric/SecretKeyManager.java

```diff
@@ -109,12 +109,13 @@ public boolean isInitialized() {
    *
    * @return true if rotation actually happens, false if it doesn't.
    */
-  public synchronized boolean checkAndRotate() throws TimeoutException {
+  public synchronized boolean checkAndRotate(boolean force)
+      throws TimeoutException {
     // Initialize the state if it's not initialized already.
     checkAndInitialize();
 
     ManagedSecretKey currentKey = state.getCurrentKey();
-    if (shouldRotate(currentKey)) {
+    if (shouldRotate(currentKey) || force) {
```

Comment:
nit: we can reverse condition "if (force || shouldRotate(currentKey))", to avoid method call if force is true.